admin_manual/issues/general_troubleshooting.rst

```diff
@@ -294,7 +414,82 @@ Make sure to change /nextcloud to the actual subfolder your Nextcloud instance i
    files by dropping any number of leading slashes, but it does not
    do so for the use in its main configuration files.
 
-If you are running NGINX, make sure ``location = /.well-known/carddav {`` and ``location = /.well-known/caldav {`` are properly configured as described in :doc:`../installation/nginx`, adapt to use a subfolder if necessary.
+.. _service-discovery-default-well-known-configuration-label:
+
+Default handling of ``/.well-known/`` URLs by Nextcloud
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Since Nextcloud allows any app to register any service within the ``.well-known`` directory, you may want to redirect
+any request to this folder to Nextcloud.
+
+.. warning::
+   **Make sure you exclude** *well known* **services that are handled by other
+   applications,** e.g. you should *not* redirect ``/.well-known/acme-challenge`` or ``/.well-known/pki-validation/`` as they
+   are used for automated SSL certificate deployment.
+
+Also, make sure to add your ``/nextcloud/subfolder``, if your Nextcloud instance is not running in the webroot.
+
+::
+
+    <IfModule mod_rewrite.c>
+      RewriteEngine on
+      # the standard redirection rules for Nextcloud
+      RewriteRule ^/\.well-known/carddav /remote.php/dav [R=301,L]
+      RewriteRule ^/\.well-known/caldav /remote.php/dav [R=301,L]
+      RewriteRule ^/\.well-known/webfinger /index.php/.well-known/webfinger [R=301,L]
+      RewriteRule ^/\.well-known/nodeinfo /index.php/.well-known/nodeinfo [R=301,L]
+
+      # Redirect all other requests to /.well-known/ to index.php
+      # exclude /.well-known/acme-challenge/
+      RewriteCond "%{REQUEST_URI}" "!^/.well-known/acme-challenge/" [OR]
+      # exclude /.well-known/pki-validation/
+      RewriteCond "%{REQUEST_URI}" "!^/.well-known/pki-validation/" [OR]
+      # add additional excludes, as required by your server setup
+      # Use a temporary redirect here (302) in case an future service is handled outside of Nextcloud
+      RewriteRule ^/\.well-known/ /index.php/ [R=302,L]
```

Comment:
I think R=302 is wrong. Apache2 should rewrite the request internally, and Nextcloud may respond with a redirect. But Apache2 should not respond with a 302 redirect to index.php.

Reply:
Well, then the same applies to the rules above which use `R=301` and are shipped with nextcloud ...

If that should be changed into an internal redirect, this should be subject to a different PR. Here, for the sake of consistency, `R=302` should be fine.

Reply:
The CalDAV/CardDAV redirects are fine as `/remote.php/dav` is the final destination and the current behavior is a described in https://www.rfc-editor.org/rfc/rfc6764#section-5

Other requests should be handled by our WellKnownController. For the sake of consistency, the examples in our documentation should match our default implementation: https://github.com/nextcloud/server/blob/3b6a9cd23628464cd24da3d426764a8232124f33/.htaccess#L92 (rewrite, not a 302 redirect).